debug.js

```diff
@@ -17,6 +17,19 @@ d._create = function(current, states) {
         }
       }
     }
+    if (state.fn) {
```

Comment:
I had to look at the code to see that this is an invoke because it has an `fn` property. For readability I'd rather do prototype checking. At the top of the file you should be able to add:

```js
let invokePromiseType = invoke(Promise.resolve());
```

And then change your check to:

```js
if (invokePromiseType.isPrototypeOf(state)) {

}
```

That way if any types get an `fn` in the future we don't accidentally trigger this code path.

Reply:
I checked, but unfortunately, the prototype there is an object, so the check for isPrototypeOf does not work. Maybe it's better to do something this way?

At the top of the file:
```js
const promiseTypeInvoke = invoke(Promise.resolve());
const equalsIgnoreOrder = (a, b) => {
  if (a.length !== b.length) return false;
  const uniqueValues = new Set([...a, ...b]);
  for (const v of uniqueValues) {
    const aCount = a.filter(e => e === v).length;
    const bCount = b.filter(e => e === v).length;
    if (aCount !== bCount) return false;
  }
  return true;
}
```

and then:
```js
if (equalsIgnoreOrder(Object.keys(state), Object.keys(promiseTypeInvoke))) {

}
```

Reply:
My bad, the reason it doesn't work is that we don't have the prototype. Need to use getPrototypeOf, try this:

```js
let invokePromiseType = Object. getPrototypeOf (invoke(Promise.resolve()));
```

Reply:
@matthewp Thanks, missed this problem, added fix

Reply:
Looks great! Merging, I won't be able to release today, please ping me if you don't see a release with a day